lib/SILOptimizer/Mandatory/DiagnoseInfiniteRecursion.cpp

```diff
@@ -109,6 +111,14 @@ static bool hasInfinitelyRecursiveApply(SILFunction *targetFn) {
   while (!workList.empty()) {
     SILBasicBlock *curBlock = workList.pop_back_val();
 
+    // If the block is "ARC inert", meaning it traps without relevant side
+    // effects, then skip it.
+    // Note that this check happens _before_ the recursion check. This is
+    // deliberate, as we want to ignore recursive calls inside a program
+    // termination point.
+    if (isARCInertTrapBB(curBlock))
```

Comment:
Can you rename this in this commit to is program termination point? Since we are expanding the use of this I would suggest that we also rename the semantics string to something more descriptive since this is no longer used just for ARC.

Reply:
Might as well move it out of ARCAnalysis and into SILBasicBlock, eh?

Reply:
Also, I think we should include something about the lack of side effects in the name, moreso than `isProgramTerminationPoint` would...